conf/mpidistutils.py

```diff
@@ -908,13 +908,38 @@ def configure_mpi(ext, config_cmd):
       return 0;
     }
     """)
-    errmsg = "Cannot {} MPI programs. Check your configuration!!!"
+    errmsg = [
+        "Cannot {} MPI programs. Check your configuration!!!",
+        "Installing mpi4py requires a working MPI implementation.",
+    ]
+    if sys.platform == 'linux':
+        errmsg += [
+            "You may be missing the MPICH or Open MPI devel package:",
+            "* On Fedora/RHEL:",
+            "  $ sudo dnf install mpich-devel     # for MPICH",
+            "  $ sudo dnf install openmpi-devel   # for Open MPI",
+            "* On Debian/Ubuntu:",
+            "  $ sudo apt install libmpich-dev    # for MPICH",
+            "  $ sudo apt install libopenmpi-dev  # for Open MPI",
+        ]
+    if sys.platform == 'darwin':
+        errmsg += [
+            "Install MPICH or Open MPI with Homebrew or MacPorts:"
+            "  $ brew install mpich|openmpi  # Homebrew",
+            "  $ port install mpich|openmpi  # MacPorts",
+        ]
```

Comment:
Q: Why is this needed for macOS? Didn't both of them provide binary packages?
- https://formulae.brew.sh/formula/mpi4py
- https://ports.macports.org/port/py-mpi4py/details/

So same as with conda, we shouldn't need special instruction?

Reply:
Fair enough, let me add more code suggestions.

Reply:
```suggestion
    if sys.platform == 'darwin':
        errmsg += [
            "Install MPICH or Open MPI with Homebrew, MacPorts, or Conda:"
            "  $ brew install mpich|openmpi                  # Homebrew",
            "  $ port install mpich|openmpi                  # MacPorts",
            "  $ conda install -c conda-forge mpich|openmpi  # Conda",
        ]
```

Reply:
If we keep this way we could very well dump the full install instructions. This is why I don't link writing documentation: no matter how much effort you put into it, someone will always have something to say about it, it is never enough.

Reply:
Exactly. I am hoping to use this PR to justify to @yarshev that this effort really makes no sense. Read the documentation, we can't be held responsible for MPI vendors.